sentry-android-ndk/src/test/java/io/sentry/android/ndk/DebugImagesLoaderTest.kt

```diff
@@ -0,0 +1,100 @@
+package io.sentry.android.ndk
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.sentry.SentryOptions
+import io.sentry.protocol.DebugImage
+import java.lang.RuntimeException
+import kotlin.test.Test
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class DebugImagesLoaderTest {
+    private class Fixture {
+        val nativeLoader = mock<IModuleListLoader>()
+        val options = SentryOptions()
+
+        fun getSut(ndk: Boolean = true): DebugImagesLoader {
+            options.isEnableNdk = ndk
+            return DebugImagesLoader(options, nativeLoader)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `get images returns null if ndk is disabled`() {
+        val sut = fixture.getSut(false)
+
+        assertNull(sut.debugImages)
+        verify(fixture.nativeLoader, never()).moduleList
+    }
+
+    @Test
+    fun `get images returns image list if ndk is enabled`() {
+        val sut = fixture.getSut()
+        whenever(fixture.nativeLoader.moduleList).thenReturn(arrayOf())
+
+        assertNotNull(sut.debugImages)
+        verify(fixture.nativeLoader).moduleList
+    }
+
+    @Test
+    fun `get images returns cached list if alreadu called`() {
+        val sut = fixture.getSut()
+        whenever(fixture.nativeLoader.moduleList).thenReturn(arrayOf())
+        assertNotNull(sut.debugImages)
+
+        whenever(fixture.nativeLoader.moduleList).thenReturn(arrayOf(DebugImage()))
+        assertTrue(sut.debugImages!!.isEmpty())
+    }
+
+    @Test
+    fun `get images cache values if ndk is enabled`() {
+        val sut = fixture.getSut()
+        whenever(fixture.nativeLoader.moduleList).thenReturn(arrayOf())
+
+        sut.debugImages
+
+        assertNotNull(sut.cachedDebugImages)
```

Comment:
`m`: I think `getCachedDebugImages` doesn't need to be `@VisibleForTesting`. The `nativeLoader` could return `debugImages` on the second invocation `whenever(fixture.nativeLoader.moduleList).thenReturn(arrayOf(/*something*/))` and we could make sure that `sut.debugImages` is still the same as in the first call.

Reply:
maybe not for this test, true, but others would still be necessary I think, if I want to test that the caches are clean, I won't be able to assert it but only `verify, never()` loads again, its still package-private so its not a big issue. making it public would be a NoGo